### PR TITLE
wasm unknown unknown support

### PR DESCRIPTION
**Pull request by vaporos**
*Updated:*

This PR allows wasm compiles using --no-default-features:

```
cargo build --target wasm32-unknown-unknown --no-default-features
```

It also works when the handler feature is enabled, which would be common for smart contracts:

```
cargo build --target wasm32-unknown-unknown --no-default-features --features=handler
```

### CODE REVIEW

**Review by agunde406 — APPROVED** [review]

File: libtransact/Cargo.toml [hunk 1]
```diff
@@ -30,12 +30,12 @@ repository = "http://github.com/hyperledger/transact"
...
 bzip2 = { version = "0.3", optional = true }
 cbor-codec = { version = "0.7", optional = true }
-cylinder = "0.2"
+cylinder = { version = "0.2", optional = true }
 glob = { version = "0.3", optional = true }
 hex = "0.3"
 libc = ">=0.2.35"
 lmdb-zero = ">=0.4.1"
-log = { version = "0.4", features = ["std"] }
+log = { version = "0.4", optional = true, features = ["std"] }
```
Comment (peterschwarz) [inline libtransact/Cargo.toml:38]: This should also be mentioned in the commit message

**Review by peterschwarz — APPROVED** [review]

File: libtransact/Cargo.toml (hunk 1 above)
Comment (vaporos) [reply libtransact/Cargo.toml:38]: Fixed

**Comment by vaporos** [comment]
Added a commit which fixes the handler feature.

File: libtransact/src/protocol/batch.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /*
  * Copyright 2018 Bitwise IO, Inc.
+ * Copyright 2021 Cargill Incorporated
```
Comment (agunde406) [inline libtransact/src/protocol/batch.rs:3]: copyright updates should be added to the other commit that updates all of the files touched in this PR

**Review by agunde406 — CHANGES_REQUESTED** [review]

File: libtransact/src/protocol/batch.rs (hunk 2 above)
Comment (agunde406) [reply libtransact/src/protocol/batch.rs:3]: They are included in the handler fix commit

Comment (vaporos) [reply libtransact/src/protocol/batch.rs:3]: This was intentional because the copyright addition is new instead of being an update.

**Review by agunde406 — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Pull request merged**